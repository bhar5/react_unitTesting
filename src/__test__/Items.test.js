import React from 'react';
import renderer from 'react-test-renderer';

import Items from '../Items';

describe("Items", () => {
  it("Default Unit Testing", () => {
    let tree = renderer.create(<Items />);
    expect(tree.toJSON()).toMatchSnapshot();
  })

  it('renders correctly when there are no items', () => {
    const tree = renderer.create(<Items />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there is one item', () => {
    const items = ['one'];
    const tree = renderer.create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when there are multiple items', () => {
    const items = ['one', 'two', 'three'];
    const tree = renderer.create(<Items items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
