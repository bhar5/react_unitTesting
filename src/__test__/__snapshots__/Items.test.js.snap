// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Default Unit Testing 1`] = `
<span>
  No items in list
</span>
`;

exports[`Items renders correctly when there are multiple items 1`] = `
<div>
  <ul>
    <li>
      one
    </li>
    <li>
      two
    </li>
    <li>
      three
    </li>
  </ul>
</div>
`;

exports[`Items renders correctly when there are no items 1`] = `
<span>
  No items in list
</span>
`;

exports[`Items renders correctly when there is one item 1`] = `
<span>
  one
</span>
`;
